crates/bevy_scene/src/scene_spawner.rs

```diff
@@ -18,8 +18,8 @@ use uuid::Uuid;
 /// See also [`SceneSpawner::instance_is_ready`].
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Event)]
 pub struct SceneInstanceReady {
```

Comment:
Just a minor observation (no pun intended 😅) unrelated to the PR changes: I see that `Trigger` is defined like this:

`pub struct Trigger<'w, E, B: Bundle = ()>`

Yet `E` is not required to be an `Event`. Wondering if this an oversight or was it on purpose?

Would `Trigger<SceneInstanceReady>` still work if `SceneInstanceReady` doesn't derive `Event`?

`World` method signatures are taking generic params or function params like this:

```rust
pub fn observe<E: Event, B: Bundle, M>
pub fn trigger(&mut self, event: impl Event)
```

So maybe adding this constraint would be correct?

I'll create a PR with said change but still wanted to write it down somewhere so I don't forget.

Reply:
On a related note: `Event` is just a marker trait. So, assuming that the `EventReader<>` functionality will continue to coexist with `Trigger<>`, I would have thought that "classic" events and observer events should use separate marker traits.